pkg/reconciler/serverlessservice/resources/services.go

```diff
@@ -98,14 +108,18 @@ func filterSubsetPorts(targetPort int32, subsets []corev1.EndpointSubset) []core
 	}
 	ret := make([]corev1.EndpointSubset, len(subsets))
 	for i, sss := range subsets {
+		org := sss.DeepCopy()
 		sst := sss.DeepCopy()
 		// Find the port we care about and remove all others.
-		for j, p := range sst.Ports {
+		for j, p := range org.Ports {
 			if p.Port == targetPort {
-				sst.Ports = sst.Ports[j : j+1]
-				break
+				sst.Ports = org.Ports[j : j+1]
+			}
+			if p.Port == networking.BackendHTTPSPort {
+				sst.Ports = append(sst.Ports, org.Ports[j:j+1]...)
 			}
```

Comment:
Is this correct if the `networking.BackendHTTPSPort` occurs before the `targetPort` in the list?

Reply:
Updated. Thank you!